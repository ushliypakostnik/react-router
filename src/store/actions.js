const fetchUrl = process.env.REACT_APP_API_URL;

export const pageToActive = (page) => ({
  type: "PAGE_TO_ACTIVE",
  activePage: page
});

export const setMinHeight = (minHeight) => ({
  type: "SET_MIN_HEIGHT",
  minHeight: minHeight
});

export const toogleTheme = (theme) => ({
  type: "TOGGLE_THEME",
  theme: theme
});

export const requestAlbums = () => ({
  type: "REQUEST_ALBUMS"
});

export const receiveAlbums = (albums) => ({
  type: "RECEIVE_ALBUMS",
  albums
});

export const requestAlbumFailed = (error) => ({
  type: "REQUEST_ALBUMS_FAILED",
  error
});

export const fetchAlbums = () => {
  return dispatch => {
    dispatch(requestAlbums());
    return fetch(fetchUrl + "/albums")
      .then(res => res.json())
      .then(
        (result) => {
          /* eslint-disable array-callback-return */
          let resultArr = [];
          result.map((item, index) => {
            if (index === 0) {
              resultArr.push({text: item.name, path: "/"});
            } else {
              resultArr.push({text: item.name, path: "/" + item.id});
            }
          });
          dispatch(receiveAlbums(resultArr));
        },
        (error) => {
          dispatch(requestAlbumFailed(error));
        }
      );
  }
}

export const requestData = () => ({
  type: "REQUEST_DATA"
});

export const receiveData = (data) => ({
  type: "RECEIVE_DATA",
  data
});

export const requestDataFailed = (error) => ({
  type: "REQUEST_DATA_FAILED",
  error
});

export const fetchData = (album) => {
  return dispatch => {
    dispatch(requestData());
    return fetch(fetchUrl + "/albums/" + album)
      .then(res => res.json())
      .then(
        (result) => {
          dispatch(receiveData(result));
        },
        (error) => {
          dispatch(requestDataFailed(error));
        }
      );
  }
}
