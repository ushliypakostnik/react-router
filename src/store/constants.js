export const FETCH_URL = process.env.REACT_APP_API_URL;

export const THEME = {
  DARK: "dark",
  LIGHT: "light"
};

export const COOKIES = {
  THEME: "theme"
};
