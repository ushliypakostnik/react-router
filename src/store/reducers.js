const reducer = (state = { }, action) => {
  switch (action.type) {
    case "REQUEST_ALBUMS":
      return Object.assign({}, state, {
        isFetching: true
      });
    case "RECEIVE_ALBUMS":
      return Object.assign({}, state, {
        isFetching: false,
        albums: action.albums
      });
    case "REQUEST_ALBUMS_FAILED":
      return Object.assign({}, state, {
        isFetching: false,
        error: action.error
      });
    case "REQUEST_DATA":
      return Object.assign({}, state, {
        isFetching: true
      });
    case "RECEIVE_DATA":
      return Object.assign({}, state, {
        isFetching: false,
        data: action.data
      });
    case "REQUEST_DATA_FAILED":
      return Object.assign({}, state, {
        isFetching: false,
        error: action.error
      });
    case "PAGE_TO_ACTIVE":
      return Object.assign({}, state, {
        activePage: action.activePage
      });
    case "TOGGLE_THEME":
      return Object.assign({}, state, {
        theme: action.theme
      });
    default:
      return state;
  }
}

export default reducer;
