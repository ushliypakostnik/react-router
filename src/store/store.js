import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { browserHistory } from "react-router";
//import { createBrowserHistory } from "history";
import { syncHistoryWithStore, routerReducer } from 'react-router-redux';

import { INITIAL_STATE } from './constants';
import { pageToActive } from './actions';
import reducer from './reducers';

const middlewares = [];
middlewares.push(thunkMiddleware)

if (process.env.NODE_ENV !== 'production') {
  const loggerMiddleware = createLogger();

  middlewares.push(loggerMiddleware);
}

function configureStore(state) {
  return createStore(
    combineReducers({
      reducer,
      routing: routerReducer
    }),
    state,
    applyMiddleware(...middlewares)
  );
}

const store = configureStore(INITIAL_STATE);

//const history = createBrowserHistory();
export const history = syncHistoryWithStore(browserHistory, store);
//history.listen(location => console.info('-> location:', location));

let lastUrl = "";
if (store.getState().routing.locationBeforeTransitions != null) {
  lastUrl = store.getState().routing.locationBeforeTransitions.pathname;
} else {
  lastUrl = "/";
}

//store.subscribe(() => {
//  console.log("Store: ", store.getState());
//  console.log("last Url: ", lastUrl);
//});

store.dispatch(pageToActive(lastUrl));

export default store;
