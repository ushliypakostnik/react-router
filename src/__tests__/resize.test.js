import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { INITIAL_STATE } from '../store/constants';

import Resize from '../containers/resize';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('resize render', () => {

  const props = {
    resize: {
      minHeight: '700px',
      deviceType: 'large'
    }
  };

  const state = Object.assign({}, INITIAL_STATE, {
    reducer: {
      resize: props.resize
    }
  });
  const store = mockStore(state);

  const wrapper = mount(<Provider store={store}>
                          <Resize {...props} />
                        </Provider>);

  it('resize render correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('resize contain wrapper with class', () => {
    expect(wrapper.find('.app__resize')).toHaveLength(1);
  });

  it('resize contain resize detector', () => {
    expect(wrapper.find('ResizeDetector')).toHaveLength(1);
  });
});
