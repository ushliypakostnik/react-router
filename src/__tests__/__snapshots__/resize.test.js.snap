// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resize render resize render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Resize)
    resize={
      Object {
        "deviceType": "large",
        "minHeight": "700px",
      }
    }
  >
    <Resize
      resize={[Function]}
    >
      <div
        className="app__resize"
      >
        <ResizeDetector
          handleHeight={true}
          handleWidth={false}
          nodeType="div"
          onResize={[Function]}
          querySelector={null}
          refreshRate={1000}
          skipOnMount={false}
          targetDomEl={null}
        >
          <ChildWrapper>
            <div />
          </ChildWrapper>
        </ResizeDetector>
      </div>
    </Resize>
  </Connect(Resize)>
</Provider>
`;
